Tests/DatadogTests/Datadog/Core/System/NetworkConnectionInfoProviderTests.swift

```diff
@@ -9,51 +9,86 @@ import Network
 @testable import Datadog
 
 class NetworkConnectionInfoProviderTests: XCTestCase {
+    @available(iOS 12, *)
     func testItStartsAndCancelsNWPathMonitor() {
-        let startExpectation = self.expectation(description: "call start")
-        let cancelExpectation = self.expectation(description: "call cancel")
+        let isStarted: (NWPathMonitor) -> Bool = { monitor in
+            return monitor.queue != nil
+        }
+
+        var pathMonitor: NWPathMonitor? = NWPathMonitor()
+        weak var weakMonitor: NWPathMonitor? = pathMonitor
+
+        XCTAssertFalse(isStarted(pathMonitor!))
+
+        var provider: NetworkConnectionInfoProvider? = NetworkConnectionInfoProvider(pathMonitor!)
 
-        let monitor = NWCurrentPathMonitorMock(pathInfo: .mockAny())
-        monitor.onStart = { startExpectation.fulfill() }
-        monitor.onCancel = { cancelExpectation.fulfill() }
+        XCTAssertTrue(isStarted(pathMonitor!))
 
         autoreleasepool {
```

Comment:
I think it will be more obvious if we capture also `var pathMonitor: NWPathMonitor? = NWPathMonitor()` (everything from line 18) within this scope.